File: android/SeatCanvasSample/app/src/main/java/com/seatcanvas/sample/BaseMapFileInfo.kt
package com.seatcanvas.sample

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

enum class ResourceScope(val value: String) {
    DEFAULT("default"),
    CUSTOMIZED("customized")
}

@Parcelize
data class BaseMapFileInfo(
    val scopeValue: String,
    val filename: String,
    val assetPath: String
) : Parcelable {
    val scope: ResourceScope
        get() = when (scopeValue) {
            ResourceScope.DEFAULT.value -> ResourceScope.DEFAULT
            ResourceScope.CUSTOMIZED.value -> ResourceScope.CUSTOMIZED
            else -> ResourceScope.DEFAULT
        }

    constructor(scope: ResourceScope, filename: String, assetPath: String) : this(
        scope.value,
        filename,
        assetPath
    )
}

File: ios/SeatCanvasSample/SeatCanvasSample/BaseMapFileInfo.swift
//
//  BaseMapFileInfo.swift
//  SeatCanvasSample
//
//  Created by king on 2026/1/20.
//

import Foundation

enum ResourceScope: String {
    case `default`
    case customized
}

class BaseMapFileInfo {
    let scope: ResourceScope
    let filename: String
    let url: URL
    init(scope: ResourceScope, url: URL) {
        self.scope = scope
        filename = url.lastPathComponent
        self.url = url
    }
}
